aws/resource_aws_appmesh_route.go

```diff
@@ -196,8 +196,9 @@ func resourceAwsAppmeshRoute() *schema.Resource {
 												},
 
 												"service_name": {
-													Type:     schema.TypeString,
-													Optional: true,
+													Type:         schema.TypeString,
+													Optional:     true,
+													RequiredWith: []string{"spec.0.grpc_route.0.match.0.method_name"},
```

Comment:
From the [API reference](https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_GrpcRouteMatch.html#appmesh-Type-GrpcRouteMatch-methodName):

> The method name to match from the request. If you specify a name, you must also specify a `serviceName`.